### ISSUE TITLE
LocalFileSystem: fs.isdir fails for links that point to directories

### ISSUE THREAD

**Issue by d-v-b** [fsspec/filesystem_spec#615]
`fs.isdir` fails to detect symlinked directories as directories. This can be fixed by checking if the destination of the link is a directory. Would this be desirable as a PR?

**Comment by martindurant** [comment]
Was this fixed in https://github.com/intake/filesystem_spec/pull/608 ?

**Comment by d-v-b** [comment]
probably... I will see if my problems go away with that change

**Comment by ianthomas23** [comment]
This was indeed fixed by #608 and can be closed.

MWE:
```python
from fsspec.implementations.local import LocalFileSystem

fs = LocalFileSystem()
fs.mkdir('/tmp/test')
fs.touch('/tmp/test/target')
fs.symlink('/tmp/test/target', '/tmp/test/link')

assert fs.islink('/tmp/test/link')
```
which is essentially the same as `test_symlink` added by that PR.
